File: 2_semester/Laba_4/task_1.py
'''
В одномерном массиве, состоящем из n вещественных элементов, вычислить:
1. Сумму отрицательных элементов.
2. Произведение элементов, расположенных между максимальным и минимальным
элементами.
Упорядочить элементы массива по возрастанию.
'''

from random import *

def generator(count, is_print):
	main_list = []
	for _ in range(count):
		main_list.append(randint(-10,10)) 
	if is_print:
		print(main_list)
	return main_list

def task_1(main_list):
	sumElemets = 0
	for element in main_list:
		if element < 0:
			sumElemets += element
	return sumElemets

def find_min(main_list):
	min = 101
	for element in main_list:
		if min > element:
			min = element
	return min

def	find_max(main_list):
	max = -101
	for element in main_list:
		if max < element:
			max = element
	return max

def find_min_element(main_list):
	min = find_min(main_list)
	for i in range(len(main_list)):
		if min == main_list[i]:
			return i

def find_max_element(main_list):
	max = find_max(main_list)
	for i in range(len(main_list)):
		if max == main_list[i]:
			return i

def task_2(main_list):
	min = find_min_element(main_list)
	max = find_max_element(main_list)
	if max > min:
		first = min
		second = max
	else:
		first = max
		second = min
	multiply = 1
	for i in range(first, second):
		multiply *= main_list[i]
	return multiply


def main():
	count = 10 #int(input("enter n: "))

	main_list = generator(count, True)

	print(task_1(main_list))

	print(task_2(main_list))



if "__main__" == __name__:
	main()

File: 2_semester/Laba_4/task_1.cpp
#include <iostream>
#include <vector>

std::vector<int> generator(int count){
	std::vector<int> main_list;
	for(int i = 0; i < count; i++){
		main_list.push_back(rand()%200-100);
	}
	return main_list;
}

int task_1(std::vector<int> main_list){
	int sumElemets = 0;
	for (auto element : main_list){
		if (element < 0){
			sumElemets += element;
		}
	}
	return sumElemets;
}
int find_min_element(std::vector<int> v){
	int min = 101;
	int min_index = -1;
	for (int i = 0; i < v.size(); i++)
		if (v[i] < min){
			min = v[i];
			min_index = i;
		}
	return min_index;
}
int find_max_element(std::vector<int> v){
	int max = -101;
	int max_index = -1;
	for (int i = 0; i < v.size(); i++)
		if (v[i] > max){
			max = v[i];
			max_index = i;
		}
	return max_index;
}

int task_2(std::vector<int> main_list){
	int max_index = find_max_element(main_list);
	int min_index = find_min_element(main_list);

	int min = (min_index < max_index) ? min_index : max_index;
	int max = (min_index < max_index) ? max_index : min_index;

	int multiply = 1;

	for (int i = min; i < max; i++){
		multiply *= main_list[i];
	}
	return multiply;
}

void task_3(std::vector<int> &main_list){
	int temp;
	for (int i = 0; i < main_list.size(); i++)
		for (int j = i; j < main_list.size(); j++)
			if (main_list[i] < main_list[j]){
				temp = main_list[i];
				main_list[i] = main_list[j];
				main_list[j] = temp;
			}
}


int main(){
	int count = 10;
	std::vector<int> main_list;
	//std::cin >> count;	

	main_list = generator(count);
	for (auto element : main_list){
		std::cout << element << " ";
	}
	std::cout << std::endl;

	std::cout << task_1(main_list) << std::endl;

	std::cout << task_2(main_list) << std::endl;

	task_3(main_list);
	for (auto element : main_list){
		std::cout << element << " ";
	}
	std::cout << std::endl;

}